src/App.tsx

```diff
@@ -1,10 +1,83 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+import { FilterType } from './types/FilterType';
+
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 11200;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [value, setValue] = useState('');
+  const completedTodos = todos.filter(todo => todo.completed);
+  const notCompletedTodos = todos.filter(todo => !todo.completed);
+
+  useEffect(() => {
+    async function fetchTodos() {
+      try {
+        const data = await getTodos(USER_ID);
+
+        setTodos(data);
+      } catch (error) {
+        setErrorMessage(ErrorType.Server);
+      }
+    }
+
+    fetchTodos();
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  const updatedTodos = (newTodo: Todo) => {
+    setTodos(prevTodos => [...prevTodos, newTodo]);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+
+      if (event.target.value) {
+        updatedTodos({
+          id: +new Date(),
+          userId: USER_ID,
+          title: event.target.value,
+          completed: false,
+        });
+      } else {
+        setErrorMessage(ErrorType.Validation);
+      }
+
+      setValue('');
+    }
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    switch (filter) {
+      case FilterType.Active:
+        return !todo.completed;
+      case FilterType.Completed:
+        return todo.completed;
+      default:
+        return todo;
+    }
+  });
```

Comment:
No mistake on this task, but it future do not filter in case FilterType is All
Also filter callback should return boolean - you have object instead

Reply:
Got it, thank you!